src/filter_iterator.rs

```diff
@@ -1,25 +1,50 @@
+use std::sync::Arc;
+
 use crate::iter::KeyValueIterator;
 use crate::types::RowEntry;
 use crate::utils::is_not_expired;
 use crate::SlateDBError;
 
+pub(crate) struct FilterIteratorPredicateBuilder {
+    ttl_now: Option<i64>,
+    max_seq: Option<u64>,
+}
+
+impl FilterIteratorPredicateBuilder {
+    pub(crate) fn new(ttl_now: Option<i64>, max_seq: Option<u64>) -> Self {
```

Comment:
Very nitty, but wdyt about `set_ttl` and `set_max_seq` to make it more builder-like?

Reply:
with @ableegoldman 's suggestion, i am afraid it might not worth to introduce this preciate builder here.. `max_seq` is filtered in the source iterators, while `ttl` is better to be filtered in the final result.

let me think another style over these preciates.

Reply:
removed this builder in 3be774f